frontend/viewer/src/lib/services/event-bus.ts

```diff
@@ -41,12 +41,13 @@ export class EventBus {
 
   public onEntryUpdated(projectName: string, callback: (entry: IEntry) => void): () => void {
     // eslint-disable-next-line func-style
-    const onEventCallback = (event: IFwEvent) => {
-      const projectEvent = event as IProjectEvent;
-      if (projectEvent.project.name !== projectName) return;
-      if (projectEvent.event.type !== FwEventType.EntryChanged) return;
+    const onEventCallback = (event: IFwEvent | IProjectEvent) => {
+      if (!('project' in event)) return;
```

Comment:
we should be able to check the event type is `FwEventType.ProjectEvent` but this is a good check too

Reply:
Refactored to use cleaner type guards.